src/components/xDragBoxLayer.ts

```diff
@@ -45,7 +45,9 @@ export module Components {
       throw new Error("yScales cannot be set on an XDragBoxLayer");
     }
 
-    public yExtent(): (number | { valueOf(): number })[] {
+    public yExtent(): (number | { valueOf(): number })[];
+    public yExtent(yExtent: (number | { valueOf(): number })[]): SelectionBoxLayer;
+    public yExtent(yExtent?: (number | { valueOf(): number })[]): any {
       throw new Error("XDragBoxLayer has no yExtent");
```

Comment:
Note that this message has not been added in this pull request, and thus this is an outside concern.